### ISSUE TITLE
Wifi is slow after upgrade to kernel 6.0.2

### ISSUE THREAD

**Issue by bengunter** [pop-os/pop#2665]
**Distribution (run `cat /etc/os-release`):**
```
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os 
```

**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
```
linux-image-6.0.2-76060002-generic:
  Installed: 6.0.2-76060002.202210150739~1666289067~22.04~fe0ce53
  Candidate: 6.0.2-76060002.202210150739~1666289067~22.04~fe0ce53
  Version table:
 *** 6.0.2-76060002.202210150739~1666289067~22.04~fe0ce53 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**

Wifi is slow (~20 Mbps) since upgrade to kernel 6.0.2.

**Steps to reproduce (if you know):**

1. Upgrade kernel
2. Suffer

**Expected behavior:**

Wifi should be as fast (300 Mbps+) as previous kernels.

**Other Notes:**

```
$ lspci
00:00.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir/Cezanne Root Complex
00:00.2 IOMMU: Advanced Micro Devices, Inc. [AMD] Renoir/Cezanne IOMMU
00:01.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge
00:01.2 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir/Cezanne PCIe GPP Bridge
00:02.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge
00:02.2 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir/Cezanne PCIe GPP Bridge
00:08.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge
00:08.1 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir Internal PCIe GPP Bridge to Bus
00:08.2 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir Internal PCIe GPP Bridge to Bus
00:14.0 SMBus: Advanced Micro Devices, Inc. [AMD] FCH SMBus Controller (rev 51)
00:14.3 ISA bridge: Advanced Micro Devices, Inc. [AMD] FCH LPC Bridge (rev 51)
00:18.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 0
00:18.1 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 1
00:18.2 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 2
00:18.3 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 3
00:18.4 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 4
00:18.5 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 5
00:18.6 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 6
00:18.7 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 7
01:00.0 Non-Volatile memory controller: SK hynix Gold P31 SSD
02:00.0 Network controller: Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter (rev 32)
03:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Lucienne (rev c1)
03:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Renoir Radeon High Definition Audio Controller
03:00.2 Encryption controller: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) Platform Security Processor
03:00.3 USB controller: Advanced Micro Devices, Inc. [AMD] Renoir/Cezanne USB 3.1
03:00.4 USB controller: Advanced Micro Devices, Inc. [AMD] Renoir/Cezanne USB 3.1
03:00.5 Multimedia controller: Advanced Micro Devices, Inc. [AMD] Raven/Raven2/FireFlight/Renoir Audio Processor (rev 01)
03:00.6 Audio device: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) HD Audio Controller
04:00.0 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 81)
04:00.1 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 81)
```

```
$ lsmod
Module                  Size  Used by
binfmt_misc            24576  1
tls                   122880  0
ccm                    20480  6
xt_nat                 16384  2
xt_tcpudp              20480  6
veth                   32768  0
xt_conntrack           16384  1
nft_chain_nat          16384  3
xt_MASQUERADE          20480  3
nf_nat                 49152  3 xt_nat,nft_chain_nat,xt_MASQUERADE
nf_conntrack_netlink    49152  0
nf_conntrack          176128  5 xt_conntrack,nf_nat,xt_nat,nf_conntrack_netlink,xt_MASQUERADE
nf_defrag_ipv6         24576  1 nf_conntrack
nf_defrag_ipv4         16384  1 nf_conntrack
xfrm_user              45056  1
xfrm_algo              16384  1 xfrm_user
xt_addrtype            16384  2
nft_compat             20480  14
nf_tables             278528  101 nft_compat,nft_chain_nat
nfnetlink              20480  4 nft_compat,nf_conntrack_netlink,nf_tables
br_netfilter           32768  0
rfcomm                 86016  4
bridge                327680  1 br_netfilter
snd_seq_dummy          16384  0
snd_hrtimer            16384  1
stp                    16384  1 bridge
llc                    16384  2 bridge,stp
overlay               163840  1
cmac                   16384  2
algif_hash             16384  1
algif_skcipher         16384  1
af_alg                 32768  6 algif_hash,algif_skcipher
bnep                   28672  2
snd_ctl_led            24576  0
snd_soc_dmic           16384  1
snd_acp3x_rn           16384  4
snd_acp3x_pdm_dma      16384  1
snd_hda_codec_realtek   163840  1
snd_sof_amd_renoir     16384  0
snd_hda_codec_generic   102400  1 snd_hda_codec_realtek
snd_sof_amd_acp        45056  1 snd_sof_amd_renoir
snd_sof_pci            24576  1 snd_sof_amd_renoir
snd_hda_codec_hdmi     81920  1
snd_sof               245760  3 snd_sof_amd_acp,snd_sof_pci,snd_sof_amd_renoir
snd_hda_intel          53248  10
snd_sof_utils          20480  1 snd_sof
snd_intel_dspcfg       36864  2 snd_hda_intel,snd_sof
snd_intel_sdw_acpi     20480  1 snd_intel_dspcfg
intel_rapl_msr         20480  0
joydev                 32768  0
intel_rapl_common      40960  1 intel_rapl_msr
snd_soc_core          372736  4 snd_acp3x_rn,snd_sof,snd_soc_dmic,snd_acp3x_pdm_dma
snd_hda_codec         172032  4 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec_realtek
snd_compress           24576  1 snd_soc_core
snd_hda_core          114688  5 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek
ac97_bus               16384  1 snd_soc_core
snd_pcm_dmaengine      20480  1 snd_soc_core
snd_hwdep              20480  1 snd_hda_codec
snd_rpl_pci_acp6x      20480  0
ath10k_pci             49152  0
edac_mce_amd           36864  0
snd_acp_pci            16384  0
dell_laptop            32768  0
snd_seq_midi           20480  0
snd_seq_midi_event     16384  1 snd_seq_midi
snd_pci_acp6x          20480  0
ath10k_core           630784  1 ath10k_pci
uvcvideo              122880  0
snd_rawmidi            45056  1 snd_seq_midi
videobuf2_vmalloc      20480  1 uvcvideo
snd_pcm               155648  12 snd_sof_amd_acp,snd_hda_codec_hdmi,snd_pci_acp6x,snd_hda_intel,snd_hda_codec,snd_sof,snd_compress,snd_soc_core,snd_sof_utils,snd_hda_core,snd_acp3x_pdm_dma,snd_pcm_dmaengine
videobuf2_memops       20480  1 videobuf2_vmalloc
snd_seq                77824  9 snd_seq_midi,snd_seq_midi_event,snd_seq_dummy
videobuf2_v4l2         32768  1 uvcvideo
btusb                  61440  0
ath                    36864  1 ath10k_core
nls_iso8859_1          16384  2
dell_wmi               24576  1 dell_laptop
videobuf2_common       81920  4 videobuf2_vmalloc,videobuf2_v4l2,uvcvideo,videobuf2_memops
snd_pci_acp5x          20480  0
snd_seq_device         16384  3 snd_seq,snd_seq_midi,snd_rawmidi
btrtl                  24576  1 btusb
ledtrig_audio          16384  4 snd_ctl_led,snd_hda_codec_generic,dell_wmi,dell_laptop
btbcm                  24576  1 btusb
kvm_amd               172032  0
snd_timer              40960  3 snd_seq,snd_hrtimer,snd_pcm
btintel                40960  1 btusb
dell_smbios            28672  2 dell_wmi,dell_laptop
btmtk                  16384  1 btusb
mac80211             1376256  1 ath10k_core
snd_rn_pci_acp3x       20480  0
kvm                  1138688  1 kvm_amd
bluetooth             876544  36 btrtl,btmtk,btintel,btbcm,bnep,btusb,rfcomm
videodev              278528  3 videobuf2_v4l2,uvcvideo,videobuf2_common
dcdbas                 20480  1 dell_smbios
snd_acp_config         16384  3 snd_rn_pci_acp3x,snd_acp_pci,snd_sof_amd_renoir
snd                   114688  45 snd_ctl_led,snd_hda_codec_generic,snd_seq,snd_seq_device,snd_hda_codec_hdmi,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek,snd_sof,snd_timer,snd_compress,snd_soc_core,snd_pcm,snd_rawmidi
ucsi_acpi              16384  0
input_leds             16384  0
ecdh_generic           16384  1 bluetooth
snd_soc_acpi           16384  2 snd_acp_config,snd_sof_amd_renoir
dell_smm_hwmon         24576  0
bfq                    90112  8
cfg80211             1069056  3 ath,mac80211,ath10k_core
mc                     69632  4 videodev,videobuf2_v4l2,uvcvideo,videobuf2_common
ecc                    40960  1 ecdh_generic
hid_multitouch         32768  0
k10temp                16384  0
snd_pci_acp3x          20480  0
rapl                   20480  0
efi_pstore             16384  0
sparse_keymap          16384  1 dell_wmi
serio_raw              20480  0
typec_ucsi             49152  1 ucsi_acpi
wmi_bmof               16384  0
soundcore              16384  2 snd_ctl_led,snd
dell_wmi_descriptor    20480  2 dell_wmi,dell_smbios
ccp                   110592  1 kvm_amd
libarc4                16384  1 mac80211
typec                  86016  1 typec_ucsi
mac_hid                16384  0
dell_rbtn              20480  0
amd_pmc                28672  0
acpi_tad               20480  0
sch_fq_codel           24576  1
kyber_iosched          40960  1
ipmi_devintf           20480  0
ipmi_msghandler        77824  1 ipmi_devintf
msr                    16384  0
parport_pc             53248  0
ppdev                  24576  0
lp                     28672  0
parport                73728  3 parport_pc,lp,ppdev
ip_tables              32768  0
x_tables               57344  7 xt_conntrack,nft_compat,xt_tcpudp,xt_addrtype,xt_nat,ip_tables,xt_MASQUERADE
autofs4                45056  2
dm_crypt               57344  2
raid10                 65536  0
raid456               167936  0
async_raid6_recov      24576  1 raid456
async_memcpy           20480  2 raid456,async_raid6_recov
async_pq               24576  2 raid456,async_raid6_recov
async_xor              20480  3 async_pq,raid456,async_raid6_recov
async_tx               20480  5 async_pq,async_memcpy,async_xor,raid456,async_raid6_recov
xor                    24576  1 async_xor
raid6_pq              126976  3 async_pq,raid456,async_raid6_recov
libcrc32c              16384  4 nf_conntrack,nf_nat,nf_tables,raid456
raid1                  49152  0
raid0                  24576  0
multipath              20480  0
linear                 20480  0
system76_io            20480  0
system76_acpi          20480  0
amdgpu              12693504  19
iommu_v2               24576  1 amdgpu
gpu_sched              49152  1 amdgpu
drm_buddy              20480  1 amdgpu
i2c_algo_bit           16384  1 amdgpu
drm_ttm_helper         16384  1 amdgpu
ttm                    98304  2 amdgpu,drm_ttm_helper
drm_display_helper    184320  1 amdgpu
cec                    81920  1 drm_display_helper
rc_core                65536  1 cec
drm_kms_helper        200704  3 drm_display_helper,amdgpu
nvme                   49152  4
syscopyarea            16384  1 drm_kms_helper
sysfillrect            20480  1 drm_kms_helper
sysimgblt              20480  1 drm_kms_helper
nvme_core             167936  5 nvme
fb_sys_fops            16384  1 drm_kms_helper
crct10dif_pclmul       16384  1
crc32_pclmul           16384  0
polyval_clmulni        16384  0
polyval_generic        16384  1 polyval_clmulni
ghash_clmulni_intel    16384  0
aesni_intel           389120  11
crypto_simd            16384  1 aesni_intel
hid_generic            16384  0
ahci                   49152  0
xhci_pci               24576  0
cryptd                 24576  5 crypto_simd,ghash_clmulni_intel
psmouse               180224  0
drm                   589824  15 gpu_sched,drm_kms_helper,drm_display_helper,drm_buddy,amdgpu,drm_ttm_helper,ttm
i2c_piix4              28672  0
libahci                49152  1 ahci
xhci_pci_renesas       20480  1 xhci_pci
nvme_common            24576  1 nvme_core
wmi                    32768  4 dell_wmi,wmi_bmof,dell_smbios,dell_wmi_descriptor
video                  61440  2 dell_wmi,dell_laptop
i2c_hid_acpi           16384  0
i2c_hid                32768  1 i2c_hid_acpi
hid                   159744  3 i2c_hid,hid_multitouch,hid_generic
```

**Comment by bengunter** [comment]
I tried the two older `linux-firmware` packages that were available using the `Force version` option in Synaptic. Those did not help.

**Comment by bengunter** [comment]
The following commands allowed me to boot the Ubuntu kernel. Wifi is fast again.

```
# apt install --no-install-suggests --no-install-recommends linux-image-5.15.0-52-generic linux-headers-5.15.0-52 linux-modules-5.15.0-52-generic linux-modules-extra-5.15.0-52-generic
# kernelstub -v -k /boot/vmlinuz-5.15.0-52-generic -i /boot/initrd.img-5.15.0-52-generic
```

**Comment by bengunter** [comment]
Fixed with upgrade to `linux-image-6.0.3-76060003-generic`
